File: Core/Entities/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Category : BaseEntity
    {
        public string Name { get; set; }
        public bool IsApplicableForVisitors { get; set; }
        public bool NeedToConsiderDays { get; set; }
        public decimal DefaultRate { get; set; }
        public bool ConsiderDefaultRate { get; set; } = false;
        public bool CoreCategory { get; set; } = false;

        public override bool Equals(object obj)
        {
            return obj is Category category &&
                   Id == category.Id &&
                   Name == category.Name;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Id,Name);
        }
    }
}

File: WebUi/src/app/shared/models/category.ts
export class Category {
    id: number;
    createdOn?: Date = new Date();
    modfiedOn?: Date = new Date();
    name: string;
    isApplicableForVisitors: boolean = true;
    needToConsiderDays: boolean = true;
    defaultRate: number = 0;
    considerDefaultRate = false;
    coreCategory: boolean = false;
}
